streamz/core.py

```diff
@@ -1231,6 +1231,111 @@ def cb(self):
             yield self._emit(x)
 
 
+@Stream.register_api()
+class to_kafka(Stream):
+    """ Writes data in the stream to Kafka
+
+    This stream accepts a string or bytes object. Call ``flush`` to ensure all
+    messages are pushed. Responses from Kafka are pushed downstream.
+
+    Parameters
+    ----------
+    topic : string
+        The topic which to write
+    producer_config : dict
+        Settings to set up the stream, see
+        https://docs.confluent.io/current/clients/confluent-kafka-python/#configuration
+        https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        Examples:
+        bootstrap.servers: Connection string (host:port) to Kafka
+
+    Examples
+    --------
+    >>> from streamz import Stream
+    >>> ARGS = {'bootstrap.servers': 'localhost:9092'}
+    >>> source = Stream()
+    >>> kafka = source.map(lambda x: str(x)).to_kafka('test', ARGS)
+    <to_kafka>
+    >>> for i in range(10):
+    ...     source.emit(i)
+    >>> kafka.flush()
+    """
+    Result = namedtuple('Result', ['err', 'msg'])
+
+    def __init__(self, upstream, topic, producer_config, **kwargs):
+        import confluent_kafka as ck
+
+        self.topic = topic
+        self.producer = ck.Producer(producer_config)
+
+        Stream.__init__(self, upstream, ensure_io_loop=True, **kwargs)
+        self.stopped = False
+        self.polltime = 0.5
+        self.loop.add_callback(self.poll)
+
+    @gen.coroutine
+    def poll(self):
+        while True:
+            # executes callbacks for any delivered data, in this thread
+            # if no meesages were sent, nothing happens
+            self.producer.poll(0)
+            if self.stopped:
+                # kill producer?
+                break
+            yield gen.sleep(self.polltime)
+
+    def update(self, x, who=None):
+        future = gen.Future()
+
+        def _():
+            while True:
+                try:
+                    # this runs asynchronously, in C-K's thread
+                    self.producer.produce(self.topic, x, callback=self.cb)
+                    future.set_result(None)
+                    return
+                except BufferError:
+                    pass
+                except Exception as e:
+                    future.set_exception(e)
+
+        self.loop.add_callback(_)
+        return future
+
+    @gen.coroutine
+    def cb(self, err, msg):
+        yield self._emit(self.Result(err, msg.value() if msg else None))
+
+    def flush(self, timeout=-1):
+        self.producer.flush(timeout)
+
+
+@Stream.register_api()
+class to_kafka_batched(Stream):
```

Comment:
read_kafka_batched was oriented around supporting Dask.  Is this the same?

Reply:
I don't know. I can see if it works with Dask. I'm trying to fix the existing code rather than design it's purpose :|  I gather better performance was expected, in terms of messages/s with batches.